### ISSUE TITLE
[Feature Request] Jackett Search Indexer Support

### ISSUE THREAD

**Issue by zero77** [masterwok/tpb-search-android#1]
Can you please add support for [Jackett](https://github.com/Jackett/Jackett) Search indexer, it can be implemented by using [Torznab](https://github.com/Sonarr/Sonarr/wiki/Implementing-a-Torznab-indexer).

Thanks
> Jackett works as a proxy server: it translates queries from apps (Sonarr, Radarr, SickRage, CouchPotato, Mylar, DuckieTV, etc) into tracker-site-specific http queries, parses the html response, then sends results back to the requesting software. This allows for getting recent uploads (like RSS) and performing searches. Jackett is a single repository of maintained indexer scraping & translation logic - removing the burden from other apps.

![hidden_mining](https://camo.githubusercontent.com/5b560454ad9fe411276cbeec42feb576d3cd06cb/68747470733a2f2f692e696d6775722e636f6d2f3064316e6c37672e706e67)

**Comment by masterwok** [comment]
Hi @zero77 thank you for your feature request. Unfortunately this project was specifically tailored to The Pirate Bay and would require significant refactoring to generalize it for multiple indexers. This effort would need to take place in a generic project built with this goal in mind.

I'm currently tied up for the near future with other projects and work but creating a project to handle multiple indexers is something I plan on doing. When I find time to start the project, I'll take a deeper look into Jackett. I would also be open to pull requests.

**Comment by zero77** [comment]
> creating a project to handle multiple indexers is something I plan on doing.

This is already possible with Jackett, it's just a case of sending search requests with get / post using the Torznab api, the rest is taken care of by Jackett.

Example, search plygin used by qbittorrent:
https://raw.githubusercontent.com/hannsen/qbittorrent_search_plugins/master/jackett.py

**Comment by masterwok** [comment]
I see, thanks for sharing. Are you asking for a Torznab indexer for Jackett  based off this project?

**Comment by zero77** [comment]
What i mean is Jackett uses the Torznab api to communicate with torrent clients and other applications. 
Jackett then relayes the search request to each torrent website that users have enabled in Jackett.

Theirs more info on the Torznab api hear: 
https://github.com/Sonarr/Sonarr/wiki/Implementing-a-Torznab-indexer

**Comment by masterwok** [comment]
Okay, so you want an Android library that can communicate with Jackett using the Torznab API?

**Comment by zero77** [comment]
It should be easier than that, more like a simple get request in Torznab format and then formatting a json or xml response.

>   GET http://servername.com/api?t=tvsearch&apikey=xxxq=The%20Beverly%20Hillbillies&season=S01

![screenshot_2018-11-12 sonarr sonarr](https://user-images.githubusercontent.com/16563034/48353080-718ef300-e686-11e8-8694-6b224fde7ed7.png)

**Comment by masterwok** [comment]
Okay, now that looks cool! Based off the query strings you posted, I wonder if [Retrofit](https://square.github.io/retrofit/) would make this a straightforward implementation. I'm going to play around with Jackett and see if this is something I want to pursue. If I do, I'll probably create it as a separate project. I'll let you know once I decide.

**Comment by masterwok** [comment]
@zero77 After looking into this, I've decided this isn't something I want to implement. I think Jackett is impressive but if I were to implement this it wouldn't work for my use case. I did learn the following though and maybe it will help you:

1. _ResultsController.cs_ is where the query endpoints are located (Potato and Torznab actions)
2. _Torznab_ action returns XML and accepts _TorznabRequest_ as an argument
3. _Potato_ action returns JSON and accepts _TorrentPotatoRequest_ as an argument
4. To view configured trackers one can issue the following request: _http://127.0.0.1:9117/api/v2.0/indexers/all/results/torznab/api?apikey=[APIKEY]&t=indexers&configured=true_
5. To query all configured trackers one can issue the following request:
_http://localhost:9117/api/v2.0/indexers/all/results/torznab/api?apikey=[APIKEY]&q=hackers%201995_
6. The Docker container would be great to develop against locally if you don't want to build the Jackett solution from source

I still think Retrofit would be a good candidate for building a client. A simple client could be built by referencing the _TorznabRequest_ and _TorrentPotatoRequest_ models fairly quickly. It may also help to implement some functionality for getting indexer capabilities within the client as well.

Sorry I couldn't be of anymore help. I hope that info was somewhat useful and wish you luck!
